src/main/java/org/springframework/integration/kafka/listener/QueueingMessageListenerInvoker.java

```diff
@@ -90,19 +98,19 @@ public void enqueue(KafkaMessage message) {
 		}
 	}
 
-	@Override
 	public void start() {
 		this.running = true;
 	}
 
-	@Override
-	public void stop() {
+	public void stop(long stopTimeout) {
+		shutdownLatch = new CountDownLatch(1);
 		this.running = false;
```

Comment:
This is in a scenario that doesn't use Application contexts. So, stop() shouldn't return while there are messages in flight. That's just wrong.